### PR TITLE
New tariffs info

### PR DESCRIPTION
**Pull request by SomeOne35719**
- Reworked TariffsInfo entity
- added new logic for tariffs
- reworked Locations and Regions entity(LocationsTranslations and RegionTranslations droped)
- droped entity courierLocations(Limits transfered to TariffsInfo entity)
- added new logic for courier ordering popUp

### CODE REVIEW

**Review by Pavl0-Gural — APPROVED** [review]

**Review by danylohlynsky — APPROVED** [review]

**Review by Andrii-Yezenitskyi — APPROVED** [review]

File: core/src/main/java/greencity/controller/OrderController.java [hunk 1]
```diff
@@ -468,4 +447,48 @@ public ResponseEntity<HttpStatus> receivePaymentClient(
...
+     * @return {@link OrderCourierPopUpDto}
+     */
+    @ApiOperation(value = "Get tariff for location and courier")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = HttpStatuses.OK),
+        @ApiResponse(code = 400, message = HttpStatuses.BAD_REQUEST),
+        @ApiResponse(code = 401, message = HttpStatuses.UNAUTHORIZED),
+        @ApiResponse(code = 404, message = HttpStatuses.NOT_FOUND)
+    })
+    @GetMapping("/tariffinfo-for-location/{locationId}")
```
Comment (thesvider) [inline core/src/main/java/greencity/controller/OrderController.java:490]: better structure would be locations/{locationId}/tariffinfo

File: core/src/main/java/greencity/controller/OrderController.java [hunk 2]
```diff
@@ -468,4 +447,48 @@ public ResponseEntity<HttpStatus> receivePaymentClient(
...
+    public ResponseEntity<OrderCourierPopUpDto> getInfoAboutTariff(@Valid @PathVariable Long locationId) {
```
Comment (thesvider) [inline core/src/main/java/greencity/controller/OrderController.java:491]: once you have Valid annotation, please add appropriate Controller test to make sure API returns 400 on invalid input parameters/path variables

File: core/src/main/java/greencity/controller/SuperAdminController.java [hunk 3]
```diff
@@ -584,4 +494,80 @@ public ResponseEntity<HttpStatus> deleteReceivingStation(@PathVariable Long id)
...
+     * @author Yurii Fedorko
+     */
+    @ApiOperation(value = "Add new tariff")
+    @ApiResponses(value = {
+        @ApiResponse(code = 201, message = HttpStatuses.OK),
+        @ApiResponse(code = 400, message = HttpStatuses.BAD_REQUEST),
+        @ApiResponse(code = 401, message = HttpStatuses.UNAUTHORIZED),
+        @ApiResponse(code = 403, message = HttpStatuses.FORBIDDEN)
+    })
+    @PostMapping("/add-new-tariff")
```
Comment (thesvider) [inline core/src/main/java/greencity/controller/SuperAdminController.java:511]: POST /tariffs

File: core/src/main/java/greencity/controller/SuperAdminController.java [hunk 4]
```diff
@@ -584,4 +494,80 @@ public ResponseEntity<HttpStatus> deleteReceivingStation(@PathVariable Long id)
...
+     * @author Yurii Fedorko
+     */
+    @ApiOperation(value = "Edit tariffLimits by  sum of Bags")
+    @ApiResponses(value = {
+        @ApiResponse(code = 202, message = HttpStatuses.OK),
+        @ApiResponse(code = 400, message = HttpStatuses.BAD_REQUEST),
+        @ApiResponse(code = 401, message = HttpStatuses.UNAUTHORIZED),
+        @ApiResponse(code = 403, message = HttpStatuses.FORBIDDEN)
+    })
+    @PatchMapping("/setLimitsByAmountOfBags/{tariffId}")
```
Comment (thesvider) [inline core/src/main/java/greencity/controller/SuperAdminController.java:531]: PATCH /tariffs/{tariffId}/limits

File: core/src/main/java/greencity/controller/SuperAdminController.java [hunk 5]
```diff
@@ -584,4 +494,80 @@ public ResponseEntity<HttpStatus> deleteReceivingStation(@PathVariable Long id)
...
+     * @author Yurii Fedorko
+     */
+    @ApiOperation(value = "Deactivate tariff")
+    @ApiResponses(value = {
+        @ApiResponse(code = 202, message = HttpStatuses.OK),
+        @ApiResponse(code = 400, message = HttpStatuses.BAD_REQUEST),
+        @ApiResponse(code = 401, message = HttpStatuses.UNAUTHORIZED),
+        @ApiResponse(code = 403, message = HttpStatuses.FORBIDDEN)
+    })
+    @PutMapping("/deactivateTariff/{tariffId}")
```
Comment (thesvider) [inline core/src/main/java/greencity/controller/SuperAdminController.java:569]: could be DELETE /tariffs/{tariffId}

File: core/src/main/java/greencity/exception/handler/CustomExceptionHandler.java [hunk 6]
```diff
@@ -745,4 +745,19 @@ public final ResponseEntity<Object> handleAccessDeniedException(AccessDeniedExce
...
+    /**
+     * Method interceptor exception {@link TariffNotFoundException}.
+     *
+     * @param ex         Exception which should be intercepted.
+     * @param webRequest contain detail about occur exception.
+     * @return ResponseEntity which contain http status and body with message of
+     *         exception.
+     */
+    @ExceptionHandler(TariffNotFoundException.class)
+    public final ResponseEntity<Object> handleTariffNotFound(TariffNotFoundException ex, WebRequest webRequest) {
```
Comment (thesvider) [inline core/src/main/java/greencity/exception/handler/CustomExceptionHandler.java:758]: we can define single handler for all *NotFound exceptions. Seems like handler logic is the same for all.

File: dao/src/main/java/greencity/entity/order/TariffsInfo.java [hunk 7]
```diff
@@ -60,4 +47,32 @@ public class TariffsInfo {
 
     @Column(nullable = false)
     private LocalDate createdAt;
+
+    @Column(name = "min_amount_of_big_bags")
```
Comment (thesvider) [inline dao/src/main/java/greencity/entity/order/TariffsInfo.java:51]: no need to specify column name as long as field name can be resolved from column name. In this case minAmountOfBigBags can be resolved from min_amount_of_big_bags

File: service/src/main/java/greencity/mapping/FindAllInfoAboutLocationDtoMapper.java [hunk 8]
```diff
@@ -13,6 +13,9 @@
 
 @Component
 public class FindAllInfoAboutLocationDtoMapper extends AbstractConverter<Region, LocationInfoDto> {
+    private static final String UALangCode = "ua";
```
Comment (thesvider) [inline service/src/main/java/greencity/mapping/FindAllInfoAboutLocationDtoMapper.java:16]: consider extracting these codes into some constants class.

**Pull request merged**